test/unittests/commands/xmlupload/test_iiif_client.py

```diff
@@ -0,0 +1,130 @@
+from unittest.mock import Mock
+from unittest.mock import patch
+
+import pytest
+from requests import RequestException
+from requests import Response
+
+from dsp_tools.commands.xmlupload.iiif_client import IIIFUriValidatorLive
+from dsp_tools.commands.xmlupload.models.input_problems import IIIFUriProblem
+
+
+class TestIIIFMakeInfoJsonUri:
+    def test_1(self) -> None:
+        test = IIIFUriValidatorLive(
+            "https://iiif.dasch.swiss/0811/1Oi7mdiLsG7-FmFgp0xz2xU.jp2/full/837,530/0/default.jp2", True
+        )
+        assert test._make_info_json_uri() == "https://iiif.dasch.swiss/0811/1Oi7mdiLsG7-FmFgp0xz2xU.jp2/info.json"
+
+    def test_2(self) -> None:
```

Comment:
For me the main thing is so that I see exactly which has failed. When I called this in a loop in one test function, then it only displayed the first failure. Usually I do make precise names, but these strings are copied from the official webpage, so I did not investigate them individually, they just must pass...

Reply:
There has to be a way of properly parametrizing pytest tests, so that you can provide it with a list and then it tests all the inputs and not just until the first failure. Normally, I don't like parametrized tests because they don't give you a way of naming each case precisely. But if that's not something that makes sense here anyway, then it might be worth investigating, how parametrizing tests can be done right.

Reply:
parametrizing is done with `@pytest.mark.parametrize`. There are already some examples in this codebase, you can simply make a repo-wide search and you'll see how to do it